src/runtime/runtime_uefi.go

```diff
@@ -0,0 +1,273 @@
+//go:build uefi
+
+package runtime
+
+import (
+	"device/x86"
+	"machine/uefi"
+)
+
+//go:linkname systemTable machine/uefi.systemTable
+var systemTable *uefi.EFI_SYSTEM_TABLE
+
+//go:linkname imageHandle machine/uefi.imageHandle
+var imageHandle uintptr
+
+func machineTicks() uint64 {
+	return uefi.Ticks()
+}
+
+type timeUnit uint64
+
+// ticks returns the number of ticks (microseconds) elapsed since power up.
+func ticks() timeUnit {
+	t := machineTicks()
+	return timeUnit(t)
+}
+
+func ticksToNanoseconds(ticks timeUnit) int64 {
+	return x86.ConvertTimeInNanoSecond(uefi.TicksFrequency(), uint64(ticks))
+}
+
+func nanosecondsToTicks(ns int64) timeUnit {
+	return timeUnit(x86.ConvertTicksFromNanoSeconds(uefi.TicksFrequency(), uint64(ns)))
+}
+
+// timeOffset
+// TODO: use rtc
+var timeOffset int64
+
+//go:linkname now time.now
+func now() (sec int64, nsec int32, mono int64) {
```

Comment:
You shouldn't need to define this function. I suggest defining the `baremetal` build tag, with which you just need to define the `ticks` function.

Reply:
x86 has Real-time-clock. This is to call the GetTime API that uses this.

Reply:
Again, you shouldn't need to define this function if you define the `baremetal` build tag. In that case, the common `time.now` function in src/runtime/baremetal.go will be used instead and you only need to implement the relevant `ticks` and `ticksToNanoseconds` functions.